src/sinks/file/mod.rs

```diff
@@ -208,7 +210,12 @@ impl FileSink {
                             debug!(message = "Closing all the open files.");
                             for (path, file) in self.files.iter_mut() {
                                 if let Err(error) = file.close().await {
-                                    error!(message = "Failed to close file.", path = ?path, %error);
+                                    emit!(&FileIoError {
+                                        error,
+                                        code: "failed_closing_file",
```

Comment:
I wonder if the `failed_` prefix on these events is redundant. It's always going to be an error or failure of some kind. Do we use any other kind of code that makes that a useful distinction?

Reply:
good point